Core/include/Acts/Vertexing/IVertexFinder.hpp

```diff
@@ -0,0 +1,40 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Any.hpp"
+#include "Acts/Utilities/Result.hpp"
+#include "Acts/Vertexing/VertexingOptions.hpp"
+
+#include <vector>
+
+namespace Acts {
+
+class Vertex;
+struct InputTrack;
+struct VertexingOptions;
+
+class IVertexFinder {
+ public:
+  using State = Acts::AnyBase<128>;
+
+  virtual Result<std::vector<Vertex>> find(
+      const std::vector<InputTrack>& trackVector,
+      const VertexingOptions& vertexingOptions, State& state) const = 0;
+
+  virtual State makeState() const = 0;
+
+  virtual bool hasTrivialState() const = 0;
```

Comment:
what does trivial mean in this context?

Reply:
Basically, this is there because the IVF has a *non-trivial* state because it needs to tell the full billoir fitter about the field cache. I'm wondering if it might be better to have the billoirfitter call take the field cache as an explicit argument, and getting rid of this extra step here.

What do you think?

Reply:
Removed in favor of adding the magnetic field context to the `makeState` function.